### ISSUE TITLE
support swagger ui

### ISSUE THREAD

**Issue by exitNA** [typora/typora-issues#3227]
the code block would be great to render openapi as swagger ui does

**Comment by jgclark** [comment]
I'd never heard of 'swagger ui' so I looked it up. From its front page I'm still not very clear what it is and particularly not clear what you're hoping Typora might implement. Please give much more information for the developers to go on.

**Comment by AlexPasternak** [comment]
Yeah, it's not a well-phrased or detailed request. However, Swagger displays APIs in a nice, easy-to-read format, so I believe the OP is asking for Typora to show the same, as shown on

https://swagger.io/tools/swagger-ui/

![Sample](https://static1.smartbear.co/swagger/media/images/tools/opensource/swagger_ui.png?ext=.png)

**Comment by exitNA** [comment]
for example:
```openapi3
tags:
- name: "pet"
  description: "Everything about your Pets"

paths:
  /pet:
    post:
      tags:
      - "pet"
      summary: "Add a new pet to the store"
      responses:
        405:
          description: "Invalid input"
    put:
      tags:
      - "pet"
      summary: "Update an existing pet"
      parameters:
      - in: query
        name: "body"
        description: "Pet object that needs to be added to the store"
        schema:
          type: string
      responses:
        400:
          description: "Invalid ID supplied"
```
will render like this:
![image](https://user-images.githubusercontent.com/5160007/74002293-c31d4f00-49a9-11ea-901c-cf4bdd8785cf.png)

**Comment by flamewave000** [comment]
This feels less like a flavour of MarkDown and more like a flavour of CSS. Swagger has a lot of styling and would bloat Typora a fair bit. Maybe if Typora built-in support for custom Plugins, swagger would be a good candidate to be a plugin.
